tests/test_fileio/test_backends/test_oss_backend.py

```diff
@@ -0,0 +1,611 @@
+                    with self.assertRaises(FileExistsError):
+                        self.backend.copytree_from_local(src, dst)
+
+        def test_copyfile_to_local(self):
+            # dst is a file
+            with tempfile.TemporaryDirectory() as tmp_dir, patch.object(
+                    self.backend, 'get', return_value=b'oss'):
+                src = os.path.join(f'{self.oss_dir}', 'file.txt')
+                dst = Path(tmp_dir) / 'file.txt'
+                self.assertEqual(self.backend.copyfile_to_local(src, dst), dst)
+                self.assertEqual(open(dst, 'rb').read(), b'oss')
+
+            # dst is a folder
+            with tempfile.TemporaryDirectory() as tmp_dir, patch.object(
+                    self.backend, 'get', return_value=b'oss'):
+                src = os.path.join(f'{self.oss_dir}', 'file.txt')
+                dst = tmp_dir
+                self.assertEqual(
+                    self.backend.copyfile_to_local(src, dst),
+                    os.path.join(dst, 'file.txt'))
+                self.assertEqual(
+                    open(os.path.join(dst, 'file.txt'), 'rb').read(), b'oss')
+
+        def test_copytree_to_local(self):
+            src = f'{self.oss_dir}'
+            dst = self.test_data_dir / 'mmengine'
+            with patch.object(
+                    self.backend, 'copytree_to_local', return_value=dst):
+                self.assertEqual(self.backend.copytree_to_local(src, dst), dst)
+
+        def test_remove(self):
+            dir_path = os.path.join(f'{self.oss_dir}', 'img.txt')
+            with patch.object(self.backend, 'remove') as remove_obj:
+                self.backend.remove(dir_path)
+                remove_obj.assert_called_once()
+
+        def test_symlink(self):
+            file_path = os.path.join(f'{self.oss_dir}', 'text5.txt')
+            linked_file_path = os.path.join(f'{self.oss_dir}',
+                                            'linked_text5.txt')
+            with self.assertRaises(FileNotFoundError):
+                self.backend.symlink(file_path, linked_file_path)
+
+else:
+    import pandas as pd
+
+    class TestOSSBackend(TestCase):  # type: ignore
+
+        @classmethod
+        def setUpClass(cls):
+            warnings.simplefilter('ignore', ResourceWarning)
+            cls.test_data_dir = Path(__file__).parent.parent.parent / 'data'
+            cls.local_img_path = cls.test_data_dir / 'color.jpg'
+            cls.local_data_dir = cls.test_data_dir / 'imgs'
+
+            cls.oss_dir = 'oss://oss-cn-hangzhou.aliyuncs.com/mmengine/'
+            local_oss_ak_path = cls.test_data_dir / 'AccessKey.csv'
+            access_key = pd.read_csv(local_oss_ak_path, sep=',')
+            cls.access_key_id = access_key.loc[0, 'AccessKey ID']
+            cls.access_key_secret = access_key.loc[0, 'AccessKey Secret']
+            print(f'access_key_id: {cls.access_key_id},\
+                access_key_secret: {cls.access_key_secret}')
```

Comment:
It would be convenient to read them from environments and then other developers can also test this unit test.

```python
cls.oss_dir = os.getenv('MMEngine_OSS_TEST_DIR')
cls.access_key_id = os.getenv('MMENGINE_OSS_ACCESS_KEY_ID')
cls.access_key_secret = os.getenv('MMENGINE_OSS_ACCESS_KEY_SECRET')
```